packages/babel-plugin/README.md

```diff
@@ -0,0 +1,239 @@
+# Realm Babel Plugin
+
+## Introduction
+
+The Realm Babel plugin enables you to define your Realm models using standard Typescript syntax - no need to define a separate schema.
+
+<table>
+<tr>
+<th>Before: without the Realm Babel plugin</th>
+<th>After: with the Realm Babel plugin</th>
+</tr>
+<tr>
+<td width="50%" valign="top">
+
+```ts
+class Task extends Realm.Object<Task, "description"> {
+  _id!: Realm.BSON.ObjectId;
+  description!: string;
+  isComplete!: boolean;
+
+  static schema = {
+    name: "Task",
+    primaryKey: "_id",
+    properties: {
+      _id: "objectId",
+      description: "string",
+      isComplete: {
+        type: "bool",
+        default: false,
+        indexed: true,
+      },
+    },
+  };
+
+  constructor(realm, description: string) {
+    super(realm, {
+      _id: new Realm.BSON.ObjectId(),
+      description,
+    });
+  }
+}
+```
+
+</td>
+<td width="50%" valign="top">
+
+```ts
+class Task extends Realm.Object<Task, "description"> {
+  _id = new Realm.BSON.ObjectId();
+  description!: string;
+  @index
+  isComplete = false;
+
+  static primaryKey = "_id";
+
+  constructor(realm, description: string) {
+    super(realm, { description });
+  }
+}
+```
+
+</code>
+</td>
+</tr>
+</table>
+
+## Features
+
+- Schema properties can be defined as class properties by using standard TypeScript types or specific `Realm.Types` types, supporting every Realm type
+- Support for default values using property initialiser syntax
+- Support for specifying additional schema properties (e.g. primary key) using class statics
+- Support for indexing and remapping fields using decorators
+
+## Installation
+
+1. Install the `@realm/babel-plugin` npm package:
+
+   `npm install --save-dev @realm/babel-plugin`
+
+2. If you don't already have it installed, install the `@babel/plugin-proposal-decorators` package (only required if you need to use the `@index` or `@mapTo` decorators):
+
+   `npm install --save-dev @babel/plugin-proposal-decorators`
+
+   and enable decorators in your `tsconfig.json` by adding: `"experimentalDecorators": true` to the `compilerOptions`.
+
+3. Update your project's `babel.config.js` to load these two plugins:
+
+   ```js
+   // Existing babel.config.js content is commented out
+   // module.exports = {
+     // presets: ['module:metro-react-native-babel-preset'],
+
+     // --------------------------
+     // Add the following plugins:
+     plugins: [
+       '@realm/babel-plugin',
+       ['@babel/plugin-proposal-decorators', { legacy: true }],
```

Comment:
I was thinking that we should also publish a `@realm/babel-config` which would check / add these plugins, to simplify consumption. Perhaps not right away, but eventually.

Reply:
That is a nice idea!

Reply:
I have created https://github.com/realm/realm-js/issues/5009 to track it